test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -1149,6 +1200,60 @@ class test_ucp_wireup_fallback_amo : public test_ucp_wireup {
     }
 };
 
+class test_ucp_wireup_amo : public test_ucp_wireup {
+public:
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = test_ucp_wireup::get_ctx_params();
+        params.field_mask  |= UCP_PARAM_FIELD_REQUEST_SIZE;
+        params.request_size = sizeof(test_ucp_wireup_amo *);
+        return params;
+    }
+
+    static std::vector<ucp_test_param>
+    enum_test_params(const ucp_params_t& ctx_params, const std::string& name,
+                     const std::string& test_case_name, const std::string& tls)
+    {
+        return enum_test_params_features(ctx_params, name, test_case_name, tls,
+                                         UCP_FEATURE_AMO32 | UCP_FEATURE_AMO64,
+                                         false);
+    }
+
+protected:
+    static void scomplete_cb(void *req, ucs_status_t status) {
```

Comment:
`scomplete_cb` = send complete callback?

Reply:
yes, copy-pasted... this test is written for review to determine firstly if it's a valid usage. If so, then there is a bug in UCX.